dsgrid/registry/registry_manager.py

```diff
@@ -221,6 +221,86 @@ def load(
         )
         return cls(params)
 
+    def data_sync(self, project_id, dataset_id):
+        """Sync data from the remote dsgrid registry.
+
+        Parameters
+        ----------
+        project_id : str
+            Sync by project_id filter
+        dataset_id : str
+            Sync by dataset_id filter
+        """
+        # TODO: docstring
+        fs_interface = self._params.fs_interface
+        cloud_interface = self._params.cloud_interface
+        offline_mode = self._params.offline
+        dry_run_mode = self._params.dry_run
+        no_prompts = True  # TODO: how do we pass in no_prompts from registry cli parent?
+
+        if not offline_mode:
+            if project_id:
+                if self.project_manager.has_id(project_id):
+                    if dataset_id:
+                        # TODO: self.dataset_manager.has_id() gives us all datasets in the registry. We want all datasets associated with a project. This is an error at the moment.
+                        if self.dataset_manager.has_id(dataset_id):
+                            version = self.dataset_manager.get_current_version(dataset_id)
+                            datasets = [(dataset_id, version)]
+                        else:
+                            raise ValueError(
+                                f"No registered dataset ID = '{dataset_id}' registered to project ID = '{project_id}'"
+                            )
+                    else:
+                        datasets = []
+                        # TODO: self.dataset_manager.has_id() gives us all datasets in the registry. We want all datasets associated with a project. This is an error at the moment.
+                        for dataset in self.dataset_manager.list_ids():
+                            version = self.dataset_manager.get_current_version(dataset)
+                            datasets.append((dataset, str(version)))
+                else:
+                    raise ValueError(f"No registered project ID = '{project_id}'")
```

Comment:
@lixiliu  see TODO comments here. Looking for the right way to filter datasets that belong to a specific project_id.

Reply:
@daniel-thom thoughts here?